docs/content/sips/xxx-manifest-redesign.md

```diff
@@ -0,0 +1,123 @@
+title = "SIP xxx - Application Manifest Redesign"
+template = "main"
+date = "2022-05-20T13:22:30Z"
+---
+
+Summary: A new design for Application Manifests (`spin.toml`)
+
+Owner: lann.martin@fermyon.com
+
+Created: May 20, 2022
+
+## Background
+
+Spin's Application Manifest file (usually named `spin.toml`) describes a Spin application's metadata,
+components, and trigger configurations. The current design supports a single trigger type and combines
+component and trigger configuration with an implicit one-to-one relationship:
+
+```toml
+name = "hello-world"
+trigger = { type = "http", base = "/hello" }
+
+[[component]]
+id = "world"
+# ...
+[component.trigger]
+route = "/world"
+```
+
+In order to support future features, we would like to enable more flexible trigger configurations:
+
+- Multiple trigger types in an application
```

Comment:
I have been wanting to have an application with multiple triggers for a long time.
However, one of the advantages of this separation is that you can clearly decide if your engine can support running the application or not — whereas this way you can be in a situation where the engine can run some components in the app manifest, but not others.

I don't have a solution, and I'm not proposing we don't do this, just opening the discussion to see what others are thinking here.

Reply:
I'm not sure I understand what you mean by:

> you can clearly decide if your engine can support running the application or not

If you can decide whether you can run _one_ type then surely you can decide whether you can run _all_ the types? :thinking:

Reply:
@radu-matei would you mind giving a concrete example?

Reply:
We have briefly talked about having specialized engines for single triggers that would be able to handle a single component type — this might be relevant in that scenario.

(node A can only handle HTTP components, while node B pub/sub components; not an issue right now, since we our current CLI implements both, but we will have to be mindful of how we handle multiple component types and nodes if we ever have such specialized engines.)

Definitely not a blocker for this PR, resolving.